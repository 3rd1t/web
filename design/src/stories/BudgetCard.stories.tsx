import React from "react";
import { BudgetCard } from '../components/BudgetCard';

export default {
  title: "Budget Card",
};

export const card = () => {

  return (
    <BudgetCard  expensesCount= {195} totalBudget={1500.00} start={new Date()} end={new Date()} budgeted={4.99} expenses={625.12} income={10.00}/>
  )
};
