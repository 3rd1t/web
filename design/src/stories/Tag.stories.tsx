import React from 'react';
import Tag from "../components/Tag"
export default { title: 'Tag' };

export const Default = () => {
    return (
        <Tag>Hello Tag</Tag>
    )
} 
