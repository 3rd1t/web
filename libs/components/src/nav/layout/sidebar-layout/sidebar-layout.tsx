import React from "react";
import { Header } from "../../top/header/header";
import TieredSidebar from "../../sidebar/tiered-sidebar/tiered-sidebar";

/* eslint-disable-next-line */
export interface SidebarLayoutProps {
  children: React.ReactNode;
  breadcrumbs?: {label: string, href: string}[];
  menuContent: {
    title: React.ReactNode;
    items: React.ReactNode[];
  }[];
  user: {
    nickname: string;
  };
}

export const SidebarLayout = ({
  children,
  breadcrumbs,
  menuContent,
  user,
}: SidebarLayoutProps) => {
  return (
    <div className="flex">
      <div>
        <div className="fixed top-0">
          <TieredSidebar menuContent={menuContent} />
        </div>
      </div>
      <div className="w-full min-h-screen pl-80">
        <div className="fixed top-0 w-full bg-white border-b border-gray-300">
          <div className="w-full">
            <Header breadcrumbs={breadcrumbs} user={user}></Header>
          </div>
        </div>
        <main className="min-h-screen pt-20 overflow-y-auto bg-warmGray-50">
          <div className="border-t border-gray-300 ">{children}</div>
        </main>
      </div>
    </div>
  );
};

export default SidebarLayout;
