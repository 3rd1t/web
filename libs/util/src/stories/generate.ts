/**
 * Return all components from storybook.
 *
 * @example
 * import * as stories from "./<component>.stories"
 *
 * const testCases = generateTestCasesFromStories(stories)
 * testCases.forEach((s) => {
 *   it(s.name + "should match snapshot", () => {
 *     const { baseElement } = render(s.component())
 *     expect(baseElement).toMatchSnapshot()
 *   })
 * })
 *
 *
 * @param stories - import * as stories from "./<component>.stories"
 * @returns Name and render function for each component
 */
export function generateTestCasesFromStories(
  stories: Record<string, any>, // eslint-disable-line
): { name: string; component: () => HTMLElement }[] {
  return Object.entries(stories)
    .filter((s) => s[0] !== "default")
    .map((s) => {
      return {
        name: s[0],
        component: s[1],
      }
    })
}
