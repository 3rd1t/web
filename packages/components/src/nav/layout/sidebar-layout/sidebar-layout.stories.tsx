import React from "react"
import { SidebarLayout } from "./sidebar-layout"

export default {
  component: SidebarLayout,
  title: "components/Navigation/Layout/SidebarLayout",
}

export const primary = () => {
  return (
    <SidebarLayout breadcrumbs={["First", "Second", "Active"]}>
      <span>Content</span>
    </SidebarLayout>
  )
}
