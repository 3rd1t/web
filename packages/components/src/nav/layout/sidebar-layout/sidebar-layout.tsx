import React from "react"
import { Header } from "../../top/header/header"
import TieredSidebar from "../../sidebar/tiered-sidebar/tiered-sidebar"

/* eslint-disable-next-line */
export interface SidebarLayoutProps {
  children: React.ReactNode
  breadcrumbs?: string[]
  menuContent: {
    title: React.ReactNode
    items: React.ReactNode[]
  }[],
  user?: {
    nickname: string
  }
}

export const SidebarLayout = ({ children, breadcrumbs, menuContent, user }: SidebarLayoutProps) => {
  return (
    <div className="flex">
      <div className="sticky top-0 flex flex-shrink-0 min-h-screen">
        <TieredSidebar menuContent={menuContent} />
      </div>
      <div className="w-full min-h-screen">
        <Header breadcrumbs={breadcrumbs} user={user}></Header>
        <main className="h-full pt-20 -mt-20 ">
          <div className="h-full bg-gray-100 border-t border-gray-300">{children}</div>
        </main>
      </div>
    </div>
  )
}

export default SidebarLayout
