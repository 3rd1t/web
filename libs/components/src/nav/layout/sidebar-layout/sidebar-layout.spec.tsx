import { render } from "@testing-library/react";
import { generateTestCasesFromStories } from "@perfolio/util/stories/generate";
import * as stories from "./sidebar-layout.stories";

describe("SidebarLayout", () => {
  const testCases = generateTestCasesFromStories(stories);
  testCases.forEach((tc) => {
    it(tc.name + " should match snapshot", () => {
      const { baseElement } = render(tc.component());
      expect(baseElement).toMatchSnapshot();
    });
  });
});
