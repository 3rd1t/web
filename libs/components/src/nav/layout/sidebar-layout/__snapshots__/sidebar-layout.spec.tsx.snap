// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarLayout primary should match snapshot 1`] = `
<body>
  <div>
    <div
      class="flex"
    >
      <div>
        <div
          class="fixed top-0"
        >
          <div
            style="width: 320px;"
          >
            <div
              class="relative"
            >
              <div
                class="absolute border-r border-gray-300"
              >
                <nav
                  class="relative flex flex-col justify-between min-h-screen pt-2 bg-gray-100 hover:shadow"
                  style="width: 80px;"
                >
                  <div>
                    <a
                      class="flex items-center justify-center w-12 h-12 m-4 focus:outline-none"
                      href="/"
                    >
                      <svg
                        class="w-10 h-10 text-gray-900 stroke-current"
                        fill="none"
                        viewBox="0 0 194 148"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M9 106.208L97.3883 17.8199L134.158 54.5894M185.07 41.8615L96.6814 130.25L59.9118 93.4803"
                          stroke-width="25"
                        />
                      </svg>
                    </a>
                    <div
                      class="mt-12"
                    >
                      <a
                        class="relative flex items-center h-12 m-4 text-gray-600 rounded focus:outline-none   bg-gradient-to-t hover:from-indigo-900 hover:to-purple-800 hover:text-white cursor-pointer hover:font-medium"
                        href="/"
                      >
                        <span
                          class="flex items-center justify-center w-12"
                        >
                          <svg
                            class="w-6 h-6"
                            fill="none"
                            stroke="currentColor"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M12 6.253v13m0-13C10.832 5.477 9.246 5 7.5 5S4.168 5.477 3 6.253v13C4.168 18.477 5.754 18 7.5 18s3.332.477 4.5 1.253m0-13C13.168 5.477 14.754 5 16.5 5c1.747 0 3.332.477 4.5 1.253v13C19.832 18.477 18.247 18 16.5 18c-1.746 0-3.332.477-4.5 1.253"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                            />
                          </svg>
                        </span>
                      </a>
                      <div
                        class="relative flex items-center h-12 m-4 text-gray-600 rounded focus:outline-none   hover:text-gray-500"
                      >
                        <span
                          class="flex items-center justify-center w-12"
                        >
                          <svg
                            class="w-6 h-6"
                            fill="none"
                            stroke="currentColor"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M9 17v-2m3 2v-4m3 4v-6m2 10H7a2 2 0 01-2-2V5a2 2 0 012-2h5.586a1 1 0 01.707.293l5.414 5.414a1 1 0 01.293.707V19a2 2 0 01-2 2z"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                            />
                          </svg>
                        </span>
                      </div>
                      <div
                        class="relative flex items-center h-12 m-4 text-gray-600 rounded focus:outline-none   hover:text-gray-500"
                      >
                        <span
                          class="flex items-center justify-center w-12"
                        >
                          <svg
                            class="w-6 h-6"
                            fill="none"
                            stroke="currentColor"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M4 4v5h.582m15.356 2A8.001 8.001 0 004.582 9m0 0H9m11 11v-5h-.581m0 0a8.003 8.003 0 01-15.357-2m15.357 2H15"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                            />
                          </svg>
                        </span>
                      </div>
                      <div
                        class="relative flex items-center h-12 m-4 text-gray-600 rounded focus:outline-none   hover:text-gray-500"
                      >
                        <span
                          class="flex items-center justify-center w-12"
                        >
                          <svg
                            class="w-6 h-6"
                            fill="none"
                            stroke="currentColor"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M19 21V5a2 2 0 00-2-2H7a2 2 0 00-2 2v16m14 0h2m-2 0h-5m-9 0H3m2 0h5M9 7h1m-1 4h1m4-4h1m-1 4h1m-5 10v-5a1 1 0 011-1h2a1 1 0 011 1v5m-4 0h4"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                            />
                          </svg>
                        </span>
                      </div>
                      <div
                        class="relative flex items-center h-12 m-4 text-gray-600 rounded focus:outline-none   hover:text-gray-500"
                      >
                        <span
                          class="flex items-center justify-center w-12"
                        >
                          <svg
                            class="w-6 h-6"
                            fill="none"
                            stroke="currentColor"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M20 7l-8-4-8 4m16 0l-8 4m8-4v10l-8 4m0-10L4 7m8 4v10M4 7v10l8 4"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                            />
                          </svg>
                        </span>
                      </div>
                    </div>
                  </div>
                  <div>
                    <div
                      class="relative flex items-center h-12 m-4 text-gray-600 rounded focus:outline-none   bg-gradient-to-t hover:from-indigo-900 hover:to-purple-800 hover:text-white cursor-pointer hover:font-medium"
                    >
                      <span
                        class="flex items-center justify-center w-12"
                      >
                        <svg
                          class="w-6 h-6"
                          fill="none"
                          stroke="currentColor"
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M14 5l7 7m0 0l-7 7m7-7H3"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                          />
                        </svg>
                      </span>
                    </div>
                    <div
                      class="relative flex items-center h-12 m-4 text-gray-600 rounded focus:outline-none   hover:text-gray-500"
                    >
                      <span
                        class="flex items-center justify-center w-12"
                      >
                        <svg
                          class="w-6 h-6"
                          fill="none"
                          stroke="currentColor"
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M12 6V4m0 2a2 2 0 100 4m0-4a2 2 0 110 4m-6 8a2 2 0 100-4m0 4a2 2 0 110-4m0 4v2m0-6V4m6 6v10m6-2a2 2 0 100-4m0 4a2 2 0 110-4m0 4v2m0-6V4"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                </nav>
              </div>
              <div
                class=""
              >
                <div
                  class="flex flex-col justify-between min-h-screen px-4 py-4 ml-20 bg-white border-r border-gray-300"
                >
                  <div
                    class="flex flex-col space-y-8"
                  >
                    <div>
                      <span
                        class="flex my-4 text-sm font-medium text-gray-600 uppercase"
                      >
                        Add Transactions
                      </span>
                      <div>
                        <a
                          aria-label="Upload statement"
                          class="flex items-center p-2 text-gray-800 rounded-lg hover:bg-gray-100"
                          href="/"
                        >
                          <span
                            class="flex items-center justify-center text-lg text-gray-400"
                          >
                            <svg
                              class="w-6 h-6"
                              fill="none"
                              stroke="currentColor"
                              viewBox="0 0 24 24"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M9 13h6m-3-3v6m5 5H7a2 2 0 01-2-2V5a2 2 0 012-2h5.586a1 1 0 01.707.293l5.414 5.414a1 1 0 01.293.707V19a2 2 0 01-2 2z"
                                stroke-linecap="round"
                                stroke-linejoin="round"
                                stroke-width="2"
                              />
                            </svg>
                          </span>
                          <div
                            class="flex items-center justify-between space-x-1"
                          >
                            <span
                              class="ml-3"
                            >
                              Upload statement
                            </span>
                          </div>
                        </a>
                      </div>
                      <div>
                        <a
                          aria-label="Add manually"
                          class="flex items-center p-2 text-gray-800 rounded-lg hover:bg-gray-100"
                          href="/app/add-transaction"
                        >
                          <span
                            class="flex items-center justify-center text-lg text-gray-400"
                          >
                            <svg
                              class="w-6 h-6"
                              fill="none"
                              stroke="currentColor"
                              viewBox="0 0 24 24"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M12 6v6m0 0v6m0-6h6m-6 0H6"
                                stroke-linecap="round"
                                stroke-linejoin="round"
                                stroke-width="2"
                              />
                            </svg>
                          </span>
                          <div
                            class="flex items-center justify-between space-x-1"
                          >
                            <span
                              class="ml-3"
                            >
                              Add manually
                            </span>
                          </div>
                        </a>
                      </div>
                    </div>
                  </div>
                  <div
                    class="pt-3 border-t border-gray-300"
                  >
                    <a
                      aria-label="Request feature"
                      class="flex items-center p-2 text-gray-800 rounded-lg hover:bg-gray-100"
                      href="mailto:request@perfol.io"
                    >
                      <span
                        class="flex items-center justify-center text-lg text-gray-400"
                      >
                        <svg
                          class="w-6 h-6 text-lightBlue-500"
                          fill="none"
                          stroke="currentColor"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M9.663 17h4.673M12 3v1m6.364 1.636l-.707.707M21 12h-1M4 12H3m3.343-5.657l-.707-.707m2.828 9.9a5 5 0 117.072 0l-.548.547A3.374 3.374 0 0014 18.469V19a2 2 0 11-4 0v-.531c0-.895-.356-1.754-.988-2.386l-.548-.547z"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                          />
                        </svg>
                      </span>
                      <div
                        class="flex items-center justify-between space-x-1"
                      >
                        <span
                          class="ml-3"
                        >
                          Request feature
                        </span>
                      </div>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="w-full min-h-screen pl-80"
      >
        <div
          class="fixed top-0 w-full bg-white border-b border-gray-300"
        >
          <div
            class="w-full"
          >
            <div
              class="z-0 flex w-full h-20"
            >
              <ul
                class="flex items-center justify-between w-full h-full px-4 pr-80"
              >
                <li
                  class="text-gray-700 "
                >
                  <div
                    class="flex items-center space-x-2"
                  >
                    <span>
                      <a
                        class="text-gray-700  hover:text-gray-500"
                      />
                    </span>
                    <span>
                      <span
                        class="font-thin text-gray-700"
                      >
                        /
                      </span>
                    </span>
                    <span>
                      <a
                        class="text-gray-700  hover:text-gray-500"
                      />
                    </span>
                    <span>
                      <span
                        class="font-thin text-gray-700"
                      >
                        /
                      </span>
                    </span>
                    <span>
                      <a
                        class="font-medium text-gradient bg-gradient-to-t from-indigo-900 to-violet-800 hover:from-indigo-600 to-violet-500"
                      />
                    </span>
                  </div>
                </li>
                <li>
                  <ul
                    class="flex items-center justify-between mr-4 space-x-16"
                  />
                </li>
              </ul>
            </div>
          </div>
        </div>
        <main
          class="min-h-screen pt-20 overflow-y-auto bg-warmGray-50"
        >
          <div
            class="border-t border-gray-300 "
          >
            <span>
              Content
            </span>
          </div>
        </main>
      </div>
    </div>
  </div>
</body>
`;
